common/zeroconf/zeroconf.h

```diff
@@ -0,0 +1,97 @@
+/*
+	Aseba - an event-based framework for distributed robot control
+	Copyright (C) 2007--2016:
+		Stephane Magnenat <stephane at magnenat dot net>
+		(http://stephane.magnenat.net)
+		and other contributors, see authors.txt for details
+	
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Lesser General Public License as published
+	by the Free Software Foundation, version 3 of the License.
+	
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU Lesser General Public License for more details.
+	
+	You should have received a copy of the GNU Lesser General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef ASEBA_ZEROCONF
+#define ASEBA_ZEROCONF
+
+#include "../types.h"
+#include "../consts.h"
+#include <map>
+#include <vector>
+#include <string>
+#include <dns_sd.h>
+
+namespace Dashel
+{
+	class Stream;
+}
+
+namespace Aseba
+{
+	/**
+	\defgroup zeroconf Zeroconf (mDNS-SD multicast DNS Service Discovery)
+		
+		Using Zeroconf, an Aseba target announces itself on the TCP/IP network
+		as a service of type _aseba._tcp, and can be discovered by clients.
+		A target might be an individual node or an Aseba switch.
+		The zeroconf service provides minimal information about a target:
+		Dashel target (host name, IP port), target name, target type, Aseba
+		protocol version, a list of node ids, and a list of node product ids.
+		A client may obtain complete node descriptions by connecting to the
+		Dashel target and requesting them using the Aseba protocol
+	*/
+	/*@{*/
+
+	class Zeroconf
+	{
+    private:
+        //! Private class that records a service of type _aseba._tcp known to the mDNS-SD daemon
+        class ZeroconfService
+        {
+        public:
+            std::string name;
+            std::string regtype;
+            std::string domain;
+            DNSServiceRef serviceref;
+        public:
+            ZeroconfService() {}
+            ZeroconfService(const std::string name, const std::string domain) : name(name), domain(domain) {}
+            ~ZeroconfService()
+            {   // release the service reference
+                if (serviceref)
```

Comment:
The DNS Service Discovery C would leave this value uninitialized, thus it could theoretically take any value, so I guess we should initialize it to zero for this test to work.